animation_nodes/nodes/gpencil/gp_layer_info.py

```diff
@@ -0,0 +1,150 @@
+]
+
+class GPLayerInfoNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_GPLayerInfoNode"
+    bl_label = "GP Layer Info"
+    errorHandlingType = "EXCEPTION"
+
+    frameType: EnumProperty(name = "Frame Type", default = "ACTIVE",
+        items = frameTypeItems, update = AnimationNode.refresh)
+
+    useFrameList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("GPLayer", "Layer", "layer", dataIsModified = True)
+
+        if self.frameType == "ACTIVE":
+            self.newInput("Scene", "Scene", "scene", hide = True)
+            self.newOutput("GPFrame", "Frame", "frame")
+            self.newOutput("Float", "Frame Number", "frameNumber")
+
+        elif self.frameType == "INDEX":
+            self.newInput(VectorizedSocket("Integer", "useFrameList",
+            ("Frame Index", "frameIndex"), ("Frame Indices", "frameIndices")))
+            self.newOutput(VectorizedSocket("GPFrame", "useFrameList",
+            ("Frame", "frame"), ("Frames", "frames")))
+            self.newOutput(VectorizedSocket("Float", "useFrameList",
+            ("Frame Number", "frameNumber"), ("Frame Numbers", "outframeNumbers")))
+
+        elif self.frameType == "FRAME":
+            self.newInput(VectorizedSocket("Float", "useFrameList",
+            ("Frame Number", "frameNumber"), ("Frame Numbers", "inframeNumbers")))
+            self.newOutput(VectorizedSocket("GPFrame", "useFrameList",
+            ("Frame", "frame"), ("Frames", "frames")))
+
+        elif self.frameType == "ALL":
+            self.newOutput("GPFrame List", "Frames", "frames")
+            self.newOutput("Float List", "Frame Numbers", "outframeNumbers")
+
+        self.newOutput("Text", "Layer Name", "layerName", hide = True)
+        self.newOutput("Text", "Layer Blend Mode", "blendMode", hide = True)
+        self.newOutput("Float", "Layer Opacity", "opacity", hide = True)
+        self.newOutput("Integer", "Layer Pass Index", "passIndex", hide = True)
+
+    def draw(self, layout):
+        layout.prop(self, "frameType", text = "")
+
+    def getExecutionFunctionName(self):
+        if self.frameType == "ACTIVE":
+            return "execute_ActiveFrame"
+        elif self.frameType == "INDEX":
+            if self.useFrameList:
+                return "execute_FrameIndices"
+            else:
+                return "execute_FrameIndex"
+        elif self.frameType == "FRAME":
+            if self.useFrameList:
+                return "execute_FrameNumbers"
+            else:
+                return "execute_FrameNumber"
+        elif self.frameType == "ALL":
+            return "execute_AllFrames"
+
+    def execute_ActiveFrame(self, layer, scene):
+        frames = layer.frames
+        if len(layer.frames) == 0:
+            return GPFrame(), 0, layer.layerName, layer.blendMode, layer.opacity, layer.passIndex
+
+        frameNumbers = self.getFrameNumbers(frames)
+        activeIndex = self.getActiveFrameIndex(scene.frame_current, frameNumbers)
```

Comment:
You should have a single `getActiveFrame` method. It returns the active frame directly. The output frame number can then be computed from the frame itself.

Reply:
Changed.